src/__tests__/customFileParserTest/integrationTest.ts

```diff
@@ -73,39 +72,31 @@ describe('customFileIntegrationTest', () => {
     if (sectorMetadata.arrayCount !== 0) {
       fileReader.readUncompressedValues();
     }
-    const geometryIndexHandlers = fileReader.readSectorGeometryIndexHandlers(fileBuffer.byteLength);
+    const compressedPrimitiveData = fileReader.readCompressedGeometryData(fileBuffer.byteLength).primitives;
 
     // Check that the sector has geometries. If it doesn't, run this test on a different file.
-    expect(geometryIndexHandlers.length).toBeGreaterThan(0);
-
-    geometryIndexHandlers.forEach(geometryIndexHandler => {
-      expect(fileGeometries.indexOf(geometryIndexHandler.name)).not.toBe(-1);
-      expect(geometryIndexHandler.indexes).toBeDefined();
-      expect(geometryIndexHandler.geometryCount).toBeGreaterThan(0);
-      expect(geometryIndexHandler.byteCount).toBeGreaterThan(0);
-      expect(geometryIndexHandler.attributeCount).toBeDefined();
+    expect(compressedPrimitiveData!.length).toBeGreaterThan(0);
+
+    compressedPrimitiveData.forEach(compressedPrimitive => {
+      expect((filePrimitiveNames).indexOf(compressedPrimitive.type)).not.toBe(-1);
+      expect(compressedPrimitive.nodeIds).toBeDefined();
+      expect(compressedPrimitive.indexes).toBeDefined();
+      expect(compressedPrimitive.count).toBeGreaterThan(0);
+      expect(compressedPrimitive.byteCount).toBeGreaterThan(0);
+      expect(compressedPrimitive.attributeCount).toBeDefined();
     });
   });
 
   test('read multi-sector file', async() => {
     const fileBuffer = fileToArrayBuffer(multiSectorFilePath);
-    const rootSector = parseManySectors(fileBuffer);
-
-    const sectors = [rootSector];
-    while (sectors.length > 0) {
-      const sector = sectors.pop();
-      expect(sector).toBeDefined();
-
-      sector!.children.forEach(child => {
-        sectors.push(child);
-      });
-
-      sector!.primitiveGroups.forEach(primitiveGroup => {
+    const { rootSector, sectors, sceneStats } = parseCustomFile(fileBuffer);
+    Object.keys(sectors).forEach(sectorId => {
```

Comment:
What is the `sectorId` here? Is it the path? If so, isn't that a better name than `sectorId`?

Reply:
No, it's an ID. I'm not sure what it's based on, but it's an actual number.